valkeyotel/metrics.go

```diff
@@ -182,7 +185,7 @@ func newClient(opts ...Option) (*otelclient, error) {
 
 func trackDialing(m dialMetrics, t dialTracer, dialFn func(context.Context, string, *net.Dialer, *tls.Config) (conn net.Conn, err error)) func(context.Context, string, *net.Dialer, *tls.Config) (conn net.Conn, err error) {
 	return func(ctx context.Context, dst string, dialer *net.Dialer, tlsConfig *tls.Config) (conn net.Conn, err error) {
-		ctx, span := t.Start(ctx, "valkey.dial", kind, trace.WithAttributes(dbattr, attribute.String("server.address", dst)), t.tAttrs)
+		ctx, span := t.Start(ctx, "valkey.dial", append(t.tAttrs, kind, trace.WithAttributes(dbattr), serverAttrs(dst))...)
```

Comment:
There's a small behavioral change here, where previously if someone was explicitly setting `server.{address,port}` by calling `otelvalkey.TraceAttrs(...)` it would have _overridden_ the erroneous `server.address` that we were setting previously.

Whereas now we will overwrite their attempted override, but since part of this PR is fixing the `server.address` attrs, I _personally_ wouldn't consider this a "breaking" change, but others may disagree (e.g. maybe someone out there is trying to intentionally vanity-plate some internal IPs, etc).

It seems unlikely that anyone who was doing this^ would be _harmed_ by this reordering, but again, anything is possible, so open to hearing what others have to say on it.

Reply:
Alright, updated to remove all of those appends. Ready for another look @rueian!